src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSourceConfig.java

```diff
@@ -29,6 +49,73 @@ protected HttpBatchSourceConfig(String referenceName) {
   @Override
   public void validate(FailureCollector failureCollector) {
     super.validate(failureCollector);
+    validateCredentials(failureCollector);
+  }
+
+  public void validateCredentials(FailureCollector collector) {
+    try {
+      if (getAuthType() == AuthType.OAUTH2) {
+        validateOAuth2Credentials(collector);
+      } else if (getAuthType() == AuthType.BASIC_AUTH) {
+        validateBasicAuthCredentials(collector);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to authenticate the given info", e);
```

Comment:
We should also call collector.addFailure in this case.

Reply:
done